diff -urN netris-0.5.org/Configure netris-0.5/Configure
--- netris-0.5.org/Configure	Sat Mar 10 10:40:52 2001
+++ netris-0.5/Configure	Sat Mar 10 10:41:05 2001
@@ -258,15 +258,4 @@
 	echo "#define CURSES_HACK" >> config.h
 fi
 
-echo "Running 'make depend'"
-if make depend; then :; else cat << END; fi
-
-make depend failed, but that's OK unless you're doing development
-END
-cat << END
-
-Now do a 'make'
-
-END
-
 # vi: ts=4 ai
diff -urN netris-0.5.org/curses.c netris-0.5/curses.c
--- netris-0.5.org/curses.c	Sat Mar 10 10:40:52 2001
+++ netris-0.5/curses.c	Sat Mar 10 10:41:25 2001
@@ -22,7 +22,8 @@
 #include "netris.h"
 #include <sys/types.h>
 #include <unistd.h>
-#include <curses.h>
+#include <ncurses/curses.h>
 #include <string.h>
 #include <stdlib.h>
+#include <time.h>
 
